h2o-core/src/main/java/water/api/NpsBinServlet.java

```diff
@@ -19,13 +20,14 @@
 
   @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response) {
-    String uri = JettyHTTPD.getDecodedUri(request);
+    JettyHTTPD jetty = H2O.getJetty();
```

Comment:
We can have a H2O specific servlet, which will provide all methods like `setResponseStatus`, `sendErrorResponse`, ..